internal/pkgecosystem/package.go

```diff
@@ -1,42 +1,50 @@
 package pkgecosystem
 
-type Pkg struct {
-	name    string
-	version string
-	manager *PkgManager
-	local   string
-	command string
+import "fmt"
+
+// Package provides an interface for working with packages from any supported package ecosystem
+type Package interface {
```

Comment:
General comment (related to all my other comments below in this file): I'm not sure if this is the right level of abstraction for this interface. It conflates a few different things together: e.g. package metadata, ecosystem specific downloading functionality, and ecosystem sandboxing. 

What we had before was actually a better separation, and perhaps what we should be doing instead is making PkgManager an interface rather than a struct. That can also be where methods like Download() go into.